File: hw/pastor/u3/hw28PrinciplesAplication/JAVASCRIPT/controller/productController.js
class ProductController {
    constructor(service) {
        this.service = service;
    }

    async addProduct(req, res) {
        try {
            const dataMap = req.body; 
            await this.service.saveNewItem(dataMap);
            res.json({ message: "Guardado correctamente" });
        } catch (error) {
            res.status(500).json({ error: error.message });
        }
    }

    async getTableData(req, res) {
        try {
            const entities = await this.service.getProcessedData();
            
            const responseData = entities.map(item => ({
                id: item._id,
                name: item.getData("name"),
                priceBase: item.getData("priceBase"),
                endPrice: item.getData("endPrice")
            }));

            res.json(responseData);
        } catch (error) {
            res.status(500).json({ error: error.message });
        }
    }
}

module.exports = ProductController;

File: hw/pastor/u3/hw28PrinciplesAplication/PYTHON/controller/product_controller.py
class ProductController:
    def __init__(self, service):
        self.service = service

    def add_product(self, data_dict):
        self.service.save_new_item(data_dict)

    def get_data_for_table(self):
        entities = self.service.get_processed_data()
        rows = []
        for entity in entities:
            rows.append((
                entity.get_id(),
                entity.get_data("name"),
                entity.get_data("priceBase"),
                entity.get_data("endPrice")
            ))
        return rows